.github/workflows/docker-publish.yml

```diff
@@ -52,6 +52,10 @@ jobs:
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
+      - name: Set output
+        id: vars
+        run: echo ::set-output name=tag::"${$(echo $env:GITHUB_REF | cut -d "/" -f3):-latest}"
```

Comment:
So there is an issue with this variable substitution, as reported by the GitHub runner. I'm not quite sure exactly what you want to do here, and I have far more experience setting up GitLab CI than GitHub ones so I don't know if there's some GitHub magic going on here.

But from what I can make out of it you want to use the GITHUB_REF if it is set and otherwise use the tag "latest"?